samza-test/src/test/java/org/apache/samza/test/processor/TestZkLocalApplicationRunner.java

```diff
@@ -0,0 +1,397 @@
+
+    // Create local application runners.
+    applicationRunner1 = new LocalApplicationRunner(applicationConfig1);
+    applicationRunner2 = new LocalApplicationRunner(applicationConfig2);
+    applicationRunner3 = new LocalApplicationRunner(applicationConfig3);
+
+    for (String kafkaTopic : ImmutableList.of(inputKafkaTopic, outputKafkaTopic)) {
+      LOGGER.info("Creating kafka topic: {}.", kafkaTopic);
+      TestUtils.createTopic(zkUtils(), kafkaTopic, 5, 1, servers(), new Properties());
+    }
+  }
+
+  @Override
+  public void tearDown() {
+    for (String kafkaTopic : ImmutableList.of(inputKafkaTopic, outputKafkaTopic)) {
+      LOGGER.info("Deleting kafka topic: {}.", kafkaTopic);
+      AdminUtils.deleteTopic(zkUtils(), kafkaTopic);
+    }
+    zkUtils.close();
+    super.tearDown();
+  }
+
+  private void publishKafkaEvents(String topic, int numEvents, String streamProcessorId) {
+    KafkaProducer producer = getKafkaProducer();
+    for (int eventIndex = 0; eventIndex < numEvents; eventIndex++) {
+      try {
+        LOGGER.info("Publish kafka event with index : {} for stream processor: {}.", eventIndex, streamProcessorId);
+        producer.send(new ProducerRecord(topic, new TestKafkaEvent(streamProcessorId, String.valueOf(eventIndex)).toString().getBytes()));
+      } catch (Exception  e) {
+        LOGGER.error("Publishing to kafka topic: {} resulted in exception: {}.", new Object[]{topic, e});
+        throw new SamzaException(e);
+      }
+    }
+  }
+
+  private ApplicationConfig buildStreamApplicationConfig(String systemName, String inputTopic,
+                                                         String processorId, String appName, String appId) {
+    Map<String, String> samzaContainerConfig = ImmutableMap.<String, String>builder()
+        .put(TaskConfig.INPUT_STREAMS(), inputTopic)
+        .put(JobConfig.JOB_DEFAULT_SYSTEM(), systemName)
+        .put(TaskConfig.IGNORED_EXCEPTIONS(), "*")
+        .put(ZkConfig.ZK_CONNECT, zkConnect())
+        .put(JobConfig.SSP_GROUPER_FACTORY(), TEST_SSP_GROUPER_FACTORY)
+        .put(TaskConfig.GROUPER_FACTORY(), TEST_TASK_GROUPER_FACTORY)
+        .put(JobCoordinatorConfig.JOB_COORDINATOR_FACTORY, TEST_JOB_COORDINATOR_FACTORY)
+        .put(JobConfig.PROCESSOR_ID(), processorId)
+        .put(ApplicationConfig.APP_NAME, appName)
+        .put(ApplicationConfig.APP_ID, appId)
+        .put(String.format("systems.%s.samza.factory", systemName), TEST_SYSTEM_FACTORY)
+        .put(JobConfig.JOB_NAME(), TEST_JOB_NAME)
+        .build();
+    Map<String, String> applicationConfig = Maps.newHashMap(samzaContainerConfig);
+    applicationConfig.putAll(StandaloneTestUtils.getKafkaSystemConfigs(systemName, bootstrapServers(), zkConnect(), null, StandaloneTestUtils.SerdeAlias.STRING, true));
+    return new ApplicationConfig(new MapConfig(applicationConfig));
+  }
+
+  @Test
+  public void shouldReElectLeaderWhenLeaderDies() throws InterruptedException {
```

Comment:
Q1: uniqueTestId in setup is used for  creating unique input/output topic per test. As long as setup, test, teardown sequence is honored, there won't be collisions. 
Q2: Yes, I will create a JIRA for unification and update it here.